pkg/agent/run.go

```diff
@@ -97,6 +98,11 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 	}
 
 	if !nodeConfig.NoFlannel {
+		if (nodeConfig.FlannelExternalIP) && (len(nodeConfig.AgentConfig.NodeExternalIPs) == 0) {
+			logrus.Warnf("Flannel-external-ip flag requires node-external-ip to be set.")
```

Comment:
Lets maybe have the message indicate what happens if the user doesn't set an external IP? I'm not sure that flannel "requires" it (in that it won't crash or exit) but it certainly won't work right without it.
```suggestion
			logrus.Warnf("Server has flannel-external-ip flag set but this node does not set node-external-ip. Flannel will use internal address when connecting to this node.")
```

Reply:
ok